src/qforte/abc/algorithm.py

```diff
@@ -62,6 +63,12 @@ class Algorithm(ABC):
 
     _res_m_evals : int
         The total number of times an individual residual element was evaluated.
+
+    _is_multi_state : bool
+        Whether to use a state-averaged approach.
+
+    _weights : list
+        List of weights in the state-averaged approach.  Defaults to an equipartition.
     """
 
     def __init__(self,
```

Comment:
This has the advantage of letting us make, e.g., PQE have multiple references with the same code

Reply:
In that case, the AnsatzAlgorithm class would allow you to have multiple references in SPQE and ADAPT-VQE without affecting the QPE and QSD. Let's not worry about it for now.